### DISCUSSION TITLE
I see that there is a flake for NixOS users

### DISCUSSION THREAD

**Discussion by refaelsh** [xmonad/xmonad#515] [Q&A] [answered]
I use Nixos, btw.
Currently I have this in my config:
```nix
    services.xserver.windowManager.xmonad = {
      enable = true;
      enableConfiguredRecompile = true;
      enableContribAndExtras = true;
      extraPackages = haskellPackages: [
        haskellPackages.xmonad-contrib
        haskellPackages.xmobar
      ];
      config = builtins.readFile ./xmonad.hs;
    };
```
Everything is working fine.
But now, I discovered that there is a flake of Xmonad here: https://github.com/xmonad/xmonad-contrib/blob/master/NIX.md.
If I use the flake, I would have something like this:
```nix
  services.xserver.windowManager.xmonad = {
    enable = true;
    enableContribAndExtras = true;
    flake = {
      enable = true;
    # prefix = "unstable";
      compiler = "ghc924";
    };
  };
```

The questions:
1. Why would I use that flake? To save 2 lines of nix code in my configuration.nix?
2. It seems that, from time-to-time, I will need to manually update the `compiler = "ghc924";`? Manually --> bad thing.
3. I cant seems to find the documentation of the NixOS module that seems provided with the flake.

Thanks in advance.

**Comment by LSLeary** [comment, marked as answer]
Directly below that example `xmonad.flake` config:

1:
> This will use core and contrib from git for your system xmonad, building your config with the compiler of your choice.

That's it. It's an easy way to use xmonad from git-master or a PR branch or your local changes, with the GHC version of your choice.

2:
> With the flake enabled, the `xmonad.haskellPackages` option is not used directly, and is instead set by the `flake.compiler` option. When compiler is unset, the default `pkgs.haskellPackages` is used.

Setting `compiler` is optional; it doesn't need updating if you don't care about it.

3:
> See the flakes themselves and nix flake documentation for full detail.

Sorry, but that `NIX.md` file and the contents of the `flake.nix` files themselves is all we got.

**Reply by refaelsh** [reply]
I understand. Thank you very much :-)
